### PR TITLE
PODS-2360 : Scheme Variances - Fix the connector in pensions-scheme to send the transformed user answers data for get scheme details

### PR DESCRIPTION
**Pull request by abhijeetardale**
PODS-2360 : Scheme Variances - Fix the connector in pensions-scheme to send the transformed user answers data for get scheme details

### CODE REVIEW

File: app/connector/SchemeConnector.scala [hunk 1]
```diff
@@ -192,4 +144,59 @@ class SchemeConnectorImpl @Inject()(
...
+  private def handleSchemeDetailsResponse(psaId: String, response: HttpResponse, url: String)(
+    implicit requestHeader: RequestHeader, executionContext: ExecutionContext) = {
+
+    val badResponseSeq = Seq("INVALID_CORRELATION_ID", "INVALID_PAYLOAD", "INVALID_IDTYPE", "INVALID_SRN", "INVALID_PSTR", "INVALID_CORRELATIONID")
+
+    response.status match {
+      case OK =>
+        val temporaryMappingTest = response.json.transform(schemeSubscriptionDetailsTransformer.transformToUserAnswers)
+
+        if (temporaryMappingTest.isSuccess)
```
Comment (grantrobson) [inline app/connector/SchemeConnector.scala]: Move this to line 177 - it's unnecessary to do this if variations switched off.

File: app/controllers/AssociatedPsaController.scala [hunk 2]
```diff
@@ -35,13 +34,17 @@ class AssociatedPsaController @Inject()(schemeConnector: SchemeConnector,
...
-
       (srn,psaId) match {
         case (Some(schemeReferenceNumber),Some(id)) =>
           schemeConnector.getSchemeDetails(id, srnRequest, schemeReferenceNumber).map {
-            case Right(schemeDetails) =>
-              val isAssociated = schemeDetails.psaDetails.exists(psa => psa.id == id)
+            case Right(json) =>
+
+              val isAssociated =  (json \ "psaDetails").asOpt[JsArray].exists(_.value.map {
+                item => item.\("id").as[String]
```
Comment (grantrobson) [inline app/controllers/AssociatedPsaController.scala]: Change to _

File: app/connector/SchemeConnector.scala (hunk 1 above)
Comment (abhijeetardale) [reply app/connector/SchemeConnector.scala]: added to get the confidence before hand for conversion

File: app/controllers/AssociatedPsaController.scala (hunk 2 above)
Comment (abhijeetardale) [reply app/controllers/AssociatedPsaController.scala]: fixed

**Pull request merged**